client/src/components/PictureModal/PictureModal.tsx

```diff
@@ -0,0 +1,137 @@
+import Typography from '@material-ui/core/Typography';
+import SaveIcon from '@material-ui/icons/Save';
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import useStyles from './PictureModalStyles';
+import React from 'react';
+import Dropzone from 'react-dropzone';
+import uploadImage from '../../helpers/APICalls/uploadImage';
+import { useState } from 'react';
+
+interface Props {
+  open: boolean;
+  setOpen: () => void;
+}
+
+interface UploadProps {
+  getRootProps: () => Event[];
+  getInputProps: () => React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
+}
+
+const PictureModal = ({ open, setOpen }: Props): JSX.Element => {
+  const classes = useStyles();
+  const maxFileSize = 5e8;
+  const acceptedFileTypes = 'image/x-png, image/png, image/jpg, image/jpeg, image/gif';
+  const [openAlert, setOpenAlert] = useState<boolean>(false);
+  const [preview, setPreview] = useState<string>('');
+  const [image, setImage] = useState<Blob>(new Blob());
+
+  const handleFile = (files: File[]) => {
+    const reader = new FileReader();
+    const currentFile = [...files].shift();
+
+    if (!currentFile) return;
+
+    reader.onload = () => {
+      const result = reader.result;
+      if (result) {
+        setPreview(result.toString());
+        setImage(currentFile);
+      }
+    };
+    reader.readAsDataURL(currentFile);
+  };
+
+  const handleUpload = async (file: Blob) => {
+    const form = new FormData();
+    form.append('image', file);
+
+    const { error } = await uploadImage(form);
+    if (error) setOpenAlert(true);
+  };
+
+  const handleAlertClose = (
+    event: React.SyntheticEvent<Event> | React.SyntheticEvent<Element, Event>,
+    reason?: SnackbarCloseReason,
+  ) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setOpenAlert(false);
+  };
+
+  const Alert = (props: AlertProps) => {
+    return <MuiAlert elevation={6} variant="filled" {...props} />;
+  };
+
+  return (
+    <Dialog open={open} onClose={setOpen}>
+      <DialogTitle disableTypography className={classes.root}>
+        <Typography variant="h6" className={classes.title}>
+          Upload profile picture
+        </Typography>
+      </DialogTitle>
+      <Divider />
+      <DialogContent>
+        <Grid container direction="column" justify="center" alignItems="center" spacing={1}>
+          <Grid item>
+            <Box className={classes.avatarBox}>
+              <Typography variant="h6">Preview</Typography>
+              <Avatar alt="preview" src={preview} className={classes.avatar} />
+            </Box>
+          </Grid>
+          <Grid item>
+            <Dropzone
+              maxSize={maxFileSize}
+              multiple={false}
+              accept={acceptedFileTypes}
+              onDrop={(acceptedFiles: File[]) => handleFile(acceptedFiles)}
+            >
+              {({ getRootProps, getInputProps }: UploadProps) => (
+                <>
```

Comment:
do we need this extra element wrapping around <Paper /> ?

Reply:
Removed it! Thanks!